### ISSUE TITLE
Add Amazon EKS Pod Identity support

### ISSUE THREAD

**Issue by jonsbun** [minio/minio-go#1940]
AWS [released](https://aws.amazon.com/about-aws/whats-new/2023/11/amazon-eks-pod-identity/) a new feature, [EKS Pod Identity](https://aws.amazon.com/blogs/aws/amazon-eks-pod-identity-simplifies-iam-permissions-for-applications-on-amazon-eks-clusters/), that aims to simplify granting AWS access to pods running in an EKS cluster. This new feature is complementary to IAM roles for service accounts (IRSA), and provides a new alternative way to securely grant AWS permissions to pods.

In short, Amazon EKS Pod Identity injects the `AWS_CONTAINER_CREDENTIALS_FULL_URI` and `AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE` environment variables into pods which currently seems like unsupported  by minio-go.

Deep dive into the new Amazon EKS Pod Identity feature: https://securitylabs.datadoghq.com/articles/eks-pod-identity-deep-dive

**Comment by harshavardhana** [comment]
We already support 

pkg/credentials/iam_aws.go:     token := os.Getenv("AWS_CONTAINER_AUTHORIZATION_TOKEN")
pkg/credentials/iam_aws.go:     fullURI := os.Getenv("AWS_CONTAINER_CREDENTIALS_FULL_URI")
pkg/credentials/iam_aws.go:     relativeURI := os.Getenv("AWS_CONTAINER_CREDENTIALS_RELATIVE_URI")

**Comment by jonsbun** [comment]
@harshavardhana `AWS_CONTAINER_CREDENTIALS_FULL_URI` is defined. However, I can't find anything about `AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE` variable.

If I am correct, minio-go supports container credential provider only for Amazon ECS, but not Amazon EKS which also requires `AWS_CONTAINER_AUTHORIZATION_TOKEN_FILE` variable.

**Comment by harshavardhana** [comment]
Feel free to add it @jonsbun
